qep-343-customized-toolbars-and-menus.md

```diff
@@ -0,0 +1,82 @@
+# QGIS Enhancement: Customized Toolbars and Menus
+
+**Date** 2025/06/24
+
+**Author** Jacky Volpes (@Djedouas)
+
+**Contact** jacky dot volpes at oslandia dot com
+
+**Version** QGIS 4.0
+
+## Summary
+
+The QGIS interface can be customized through the Settings > Options menu (e.g. to add processing algorithms to the toolbar) and through the Settings > Interface Customization dialog (deactivate/activate elements in menus, docks, toolbars, ...).
+
+In addition to activating or deactivating action items within the predefined menus and toolbars, this proposal would allow for the creation of “new” toolbars and menus, composed of an individual selection of the available items in the Interface Customization dialog.
+
+Custom toolbars can currently be created with the plugin [Customize ToolBars](https://plugins.qgis.org/plugins/CustomToolBar/). The plugin has good ratings and has been downloaded frequently, showing that there is an interest for this functionality. Limitations of this plugin are:
+
+  * the storing of the settings in a binary file outside the QGIS folder structure
+  * possible maintenance issues
+  * loading of saved configurations is not possible through the plugin GUI (less intuitive for users)
+  * it's a plugin
+
+## Proposed Solution
+
+The integration of customized toolbars within the Interface Customization dialog would be a logical follow-up to the already available GUI customization possibilities. The configuration would also be saved within the user profile QGIS > QGISCUSTOMIZATION3.ini file.
+
+Additions to the Interface Customization dialog would be:
+
+  * similarly to the attribute form settings, a drop-down that allows to use a "drag and drop designer" for the configuration of the different components of the GUI
+  * if the "drag and drop designer" is chosen, the possibility (green plus icon) to add new "containers" (similarly to adding new containers in the attribute form settings), with the choice between "menu" and "toolbar"
+  * a "provide .ini file" option to load an existing configuration
+  * an "export configuration" button opening the profile folder at QGIS > QGISCUSTOMIZATION.ini
+
```

Comment:
Do you think it will be possible to import an existing toolbar[s] (clone) to adjust it[them] to user needs?

Reply:
Good point, I think that in the UI where the user selects the tool-buttons he wants to add to his custom toolbar, he will browse existing tool-buttons and existing toolbars.

If he selects an existing toolbar, it will add all the buttons at once in his customized toolbar.

Reply:
QEP updated accordingly